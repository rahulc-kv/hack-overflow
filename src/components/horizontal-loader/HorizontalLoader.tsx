import React, { FC } from 'react';

import { HorizontalLoaderProps } from './types';

const HorizontalLoader: FC<HorizontalLoaderProps> = props => {
  const { customWidth = 'w-[50%]' } = props;
  return (
    <div
      className={`overflow-hidden relative ${customWidth} h-3 bg-white rounded-xl`}>
      <div
        className="absolute w-[200%] h-full bg-gradient-to-r from-blueBell  
    via-lightGrey to-blueBell rounded-xl animate-slide-infinite"
      />
    </div>
  );
};

export default HorizontalLoader;
