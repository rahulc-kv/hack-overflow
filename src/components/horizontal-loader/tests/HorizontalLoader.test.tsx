import React from 'react';
import { render } from '@testing-library/react';

import HorizontalLoader from '../HorizontalLoader';

test('HorizontalLoader rendered', () => {
  const component = render(<HorizontalLoader customWidth="w-[70%]" />);
  expect(component).toMatchSnapshot();
});
