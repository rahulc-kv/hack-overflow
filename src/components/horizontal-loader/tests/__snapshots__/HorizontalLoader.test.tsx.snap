// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalLoader rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overflow-hidden relative w-[70%] h-3 bg-white rounded-xl"
      >
        <div
          class="absolute w-[200%] h-full bg-gradient-to-r from-blueBell   via-lightGrey to-blueBell rounded-xl animate-slide-infinite"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overflow-hidden relative w-[70%] h-3 bg-white rounded-xl"
    >
      <div
        class="absolute w-[200%] h-full bg-gradient-to-r from-blueBell   via-lightGrey to-blueBell rounded-xl animate-slide-infinite"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
